packages/environment-check/src/checks/environment.ts

```diff
@@ -137,31 +204,44 @@ async function getDestinationsResults(
  */
 export async function checkEnvironment(options?: CheckEnvironmentOptions): Promise<EnvironmentCheckResult> {
     const logger = getLogger();
+    let destinations: Destination[];
+    let destinationResults: { [dest: string]: DestinationResults };
 
     const { environment, messages } = await getEnvironment();
     logger.push(...messages);
 
-    const deepDiveDestinations = options?.destinations ? new Set(options.destinations) : new Set<string>();
-    if (options?.workspaceRoots?.length > 0) {
-        const workspaceResults = await getDestinationsFromWorkspace(options?.workspaceRoots);
-        logger.push(...workspaceResults.messages);
-        workspaceResults.destinations.forEach((dest) => deepDiveDestinations.add(dest));
-    }
+    const markdownTitle = getMarkdownTitle();
+
+    if (isAppStudio()) {
+        const deepDiveDestinations = options?.destinations ? new Set(options.destinations) : new Set<string>();
+        if (options?.workspaceRoots?.length > 0) {
+            const workspaceResults = await getDestinationsFromWorkspace(options?.workspaceRoots);
+            logger.push(...workspaceResults.messages);
+            workspaceResults.destinations.forEach((dest) => deepDiveDestinations.add(dest));
+        }
 
-    const { messages: destMessages, destinations } = await checkBASDestinations();
-    logger.push(...destMessages);
+        const basDestResults = await checkBASDestinations();
+        destinations = basDestResults.destinations;
+        logger.push(...basDestResults.messages);
 
-    const { messages: destDetailsMessages, destinationResults } = await getDestinationsResults(
-        deepDiveDestinations,
-        destinations,
-        options?.credentialCallback
-    );
-    logger.push(...destDetailsMessages);
+        const destResults = await getDestinationsResults(
+            deepDiveDestinations,
+            destinations,
+            options?.credentialCallback
+        );
+        destinationResults = destResults.destinationResults;
+        logger.push(...destResults.messages);
+    }
+
+    const toolsExtensionResults = await getToolsExtensions();
```

Comment:
I've tested in BAS and when I enter this function I get an error:

```
Error checking environment: Error: spawn code ENOENT
```

Because command `code` doesn't exist in BAS. This indicates also that we need a better error handling, if getting one installed extension/module fails, it should continue with the others and not completely exit.

Reply:
Fixed